modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -3327,17 +3328,18 @@ private void handleError() {
             ip = -1;
             if (context.getServiceInfo() != null) {
                 // Invoke ServiceConnector error handler.
-                Object protocol = context.getCarbonMessage().getProperty("PROTOCOL");
-                Optional<ServerConnectorErrorHandler> optionalErrorHandler =
-                        BallerinaConnectorManager.getInstance().getServerConnectorErrorHandler((String) protocol);
-                try {
-                    optionalErrorHandler
-                            .orElseGet(DefaultServerConnectorErrorHandler::getInstance)
-                            .handleError(new BallerinaException(
-                                            BLangVMErrors.getPrintableStackTrace(context.getError())),
-                                    context.getCarbonMessage(), context.getBalCallback());
-                } catch (Exception e) {
-                    logger.error("cannot handle error using the error handler for: " + protocol, e);
+                CarbonMessage carbonMessage = context.getCarbonMessage();
+                if (carbonMessage != null) {
```

Comment:
This logic to invoke server connector error handler, if the execution is initiated from a client. 

If the carbon message is null, that means there is no client involved, hence we can't invoke the server connector error handler.  So we just skip. 

This is not an error condition (carbonMessage == null), as this happens at service init.  Hence it is not required to throw an error.